### PR TITLE
Refactoring application gateway

### PR DESCRIPTION
**Pull request by uolter**
@AleDore @michaeldisaro this PR has already been applied in UAT.
Can u pls check everything is working fine?
While we'll apply it in prod it might cause outages impacting the final users.

### CODE REVIEW

**Review by AleDore — APPROVED** [review]
It seems everything is working fine in UAT

File: code/public.tf [hunk 1]
```diff
@@ -20,146 +20,112 @@ locals {
...
-    protocol                                  = "Http"
-    timeout                                   = 30
-    interval                                  = 30
-    unhealthy_threshold                       = 3
-
-    match {
-      status_code = ["200-399"]
+  # SKU
+  sku_name = "WAF_v2"
+  sku_tier = "WAF_v2"
```
Comment (pasqualedevita) [inline code/public.tf]: probably here we need a variable to use standard_v2 sku in uat and WAF_v2 in prod.
Also in uat we need to explicit disable waf

File: code/public.tf [hunk 2]
```diff
@@ -20,146 +20,112 @@ locals {
...
+  app_gateway_min_capacity = var.app_gateway_min_capacity
+  app_gateway_max_capacity = var.app_gateway_max_capacity
 
-  autoscale_configuration {
-    min_capacity = var.app_gateway_min_capacity
-    max_capacity = var.app_gateway_max_capacity
-  }
+  # Logs
+  sec_log_analytics_workspace_id = null
+  sec_storage_id                 = null
```
Comment (pasqualedevita) [inline code/public.tf:130]: in prod send log to prodsec

File: code/public.tf [hunk 3]
```diff
@@ -20,146 +20,112 @@ locals {
...
-    enabled                  = true
-    firewall_mode            = "Detection"
-    rule_set_type            = "OWASP"
-    rule_set_version         = "3.1"
-    request_body_check       = true
-    file_upload_limit_mb     = 100
-    max_request_body_size_kb = 128
-  }
+  # Scaling
+  app_gateway_min_capacity = var.app_gateway_min_capacity
```
Comment (pasqualedevita) [inline code/public.tf:125]: set min capacity to 0 for UAT

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

**Comment by pasqualedevita** [comment]
Also we can add standards alerts as here https://github.com/pagopa/io-infra/blob/main/src/core/appgateway.tf#L372

File: code/public.tf (hunk 2 above)
Comment (uolter) [reply code/public.tf:130]: > 

No I will do in another PR

**Comment by uolter** [comment]
> Also we can add standards alerts as here https://github.com/pagopa/io-infra/blob/main/src/core/appgateway.tf#L372

No I will do in another PR

**Comment by uolter** [comment]
> It seems everything is working fine in UAT

Thank you !!!

**Pull request merged**